### PR TITLE
MP-3261 Code sniffer recommends removing a valid @throw

### PR DESCRIPTION
**Pull request by limeeugenia**
- Developer(s): @limeeugenia

- Ticket: https://spryker.atlassian.net/browse/MP-3261

- Academy PR: ACADEMY_URL_HERE

- Release Group: https://release.spryker.com/release-groups/view/2781

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   CodeSniffer           | patch (currently 0.x)  |                       |

-----------------------------------------

##### Change log

Initial Release

- Adjusted `DocBlockThrowsSniff` to detect static methods providing the exception preset.

### CODE REVIEW

**Comment by dereuromark** [comment]
Whats a code example that contains a valid throw that would be removed?

File: Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php [hunk 1]
```diff
@@ -104,16 +104,35 @@ protected function extractExceptions(File $phpCsFile, int $stackPointer): array
...
+                $exceptions[] = $this->extractException($phpCsFile, $contentIndex);
+
                 continue;
             }
 
-            $exceptions[] = $this->extractException($phpCsFile, $contentIndex);
+            if ($classIndex && $doubleColonIndex) {
+                $exceptions[] = [
+                    'start' => $classIndex,
+                    'end' => $classIndex+1,
```
Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php:129]: small CS issue, see failing tests

**Comment by limeeugenia** [comment]
>did you also check if Spryker cores need adjustment as merge only PRs?

I did check this locally and it only affects modules that I already plan to release. See https://github.com/spryker/spryker/pull/7293 and https://github.com/spryker/suite-nonsplit/pull/4270 (I set the code sniffer to my branch in both to have a BC check).

I also checked the auto-fix: it works as expected fo this new case.

**Comment by dereuromark** [comment]
This will be merged together with the core PRs here. No public release just yet.

File: Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php [hunk 2]
```diff
@@ -104,16 +104,35 @@ protected function extractExceptions(File $phpCsFile, int $stackPointer): array
             }
 
             $newIndex = $phpCsFile->findNext(T_NEW, $i + 1, $scopeCloser);
-            if (!$newIndex) {
+
+            $classIndex = $phpCsFile->findNext(T_STRING, $i + 1, $scopeCloser);
+            $doubleColonIndex = $phpCsFile->findNext(T_DOUBLE_COLON, $i + 1, $scopeCloser);
```
Comment (darius-telycenas) [inline Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php:109]:
```suggestion
            $doubleColonIndex = $phpCsFile->findNext(T_PAAMAYIM_NEKUDOTAYIM, $i + 1, $scopeCloser);
```

Comment (dereuromark) [reply Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php:109]: Whats the reason for this? Afaik those are identical, no?

Comment (darius-telycenas) [reply Spryker/Sniffs/Commenting/DocBlockThrowsSniff.php:109]: ;)) the reason? — sounds much **more interesting**! and also makes devs research the rich (over quarter of century…) history of PHP.

…but yeah, even `390` would also be exactly the same (just even less readable).

**Pull request merged**